File: app/src/main/java/com/example/mvvmframe/base/BaseActivity.java
package com.example.mvvmframe.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import com.example.mvvmframe.utils.ActivityManager;

/**
 * create by libo
 * create on 2020/8/4
 * description
 */
public abstract class BaseActivity<V extends ViewDataBinding, VM extends BaseViewModel> extends AppCompatActivity implements IBaseView {

    protected VM viewModel;
    private V binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initViewModel();
        initDataBinding();
        init();

        ActivityManager.getInstance().addActivity(this);  //创建Activity入栈管理
    }

    protected abstract int getLayoutId();

    private void initViewModel() {
        viewModel = getViewModel();
    }

    private void initDataBinding() {
        binding = DataBindingUtil.setContentView(this, getLayoutId());
        if (binding != null) {
            binding.executePendingBindings();
        }
    }

    protected abstract void init();

    protected abstract VM getViewModel();

    @Override
    protected void onDestroy() {
        super.onDestroy();

        ActivityManager.getInstance().removeActivity(this);  //销毁Activity移出栈
    }

    @Override
    public void showToast() {

    }

    @Override
    public void showLoading() {

    }

    @Override
    public void showEmpty() {

    }
}

File: library-base/src/main/java/com/example/base/network/base/activity/BaseActivity.kt
package com.example.base.network.base.activity

import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.example.base.R
import com.example.base.network.base.view.IBaseView
import com.example.base.network.utils.ActivityManager
import com.gyf.immersionbar.ImmersionBar

/**
 * create by libo
 * create on 2020/8/7
 * description BaseActivity
 */
open abstract class BaseActivity : AppCompatActivity(), IBaseView {

    override fun onCreate(savedInstanceState: Bundle?) {
        //统一设置activity竖屏
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        super.onCreate(savedInstanceState)

        setContentView(setLayoutId())

        ActivityManager.addActivity(this) //创建Activity入栈管理

        statusBaySetting()
    }

    /**
     * activity系统状态栏封装，并设置默认样式
     */
    open fun statusBaySetting() {
        lifecycle.addObserver(object: LifecycleObserver {
            @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
            fun onCreate() {
                ImmersionBar.with(this@BaseActivity)
                        .statusBarColor(statusBarColor())
                        .statusBarDarkFont(isDarkFont())
                        .fitsSystemWindows(true)
                        .init()
            }
        })
    }

    protected abstract fun setLayoutId(): Int

    /**
     * 设置系统状态栏颜色
     */
    open fun statusBarColor(): Int = R.color.colorPrimary

    /**
     * 设置状态栏颜色，默认黑色
     */
    open fun isDarkFont(): Boolean = true

    override fun showToast() {}

    override fun showLoading() {}

    override fun showEmpty() {}

    override fun onDestroy() {
        super.onDestroy()
        ActivityManager.removeActivity(this) //销毁Activity移出栈
    }
}